bindings/c/include/opendal.h

```diff
@@ -81,6 +81,112 @@ typedef enum opendal_code {
   OPENDAL_RANGE_NOT_SATISFIED,
 } opendal_code;
 
+/**
+ * \brief Represents the status of an asynchronous operation future.
+ */
+typedef enum opendal_future_poll_status {
```

Comment:
I'm not sure if it's a good idea to expose future AS-IS. I'd prefer that we have some actual C users first before introducing features like this.

Actually, I have no idea how C users handle async IO. Maybe we can start by discussing it.

Reply:
Agreed. Would you like to start a new discussion or issue for this (along with those wonderful resource links)?